pkg/lvm/lvm_util.go

```diff
@@ -668,6 +866,144 @@ func ListLVMLogicalVolume() ([]LogicalVolume, error) {
 	return decodeLvsJSON(output)
 }
 
+/*
+ListLVMPhysicalVolume invokes `pvs` to list all the available LVM physical volumes in the node.
+*/
+func ListLVMPhysicalVolume() ([]PhysicalVolume, error) {
+	if err := ReloadLVMMetadataCache(); err != nil {
+		return nil, err
+	}
+
+	args := []string{
+		"--options", "pv_all,vg_name",
+		"--reportformat", "json",
+		"--units", "b",
+	}
+	cmd := exec.Command(PVList, args...)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		klog.Errorf("lvm: error while running command %s %v: %v", PVList, args, err)
+		return nil, err
+	}
+	return decodePvsJSON(output)
+}
+
+//To parse the output of pvs command and store it in PhysicalVolume
+//It returns PhysicalVolume.
+//
+//Example: PhysicalVolume{
+//		Name:         "/dev/sdc",
+//      UUID:         "UAdQl0-dK00-gM1V-6Vda-zYeu-XUdQ-izs8KW"
+//		Size:         21441282048
+//		Used:         8657043456
+//		Free:         12784238592
+//		MetadataSize: 1044480
+//		MetadataFree: 518656
+//		DeviceSize:   21474836480
+//		Allocatable:  "allocatable"
+//		InUse:        "used"
+//		Missing:      ""
+//		VGName:       "vg_thin"
+//	}
+func parsePhysicalVolume(m map[string]string) (PhysicalVolume, error) {
+	var pv PhysicalVolume
+	var err error
+	var sizeBytes int64
+
+	pv.Name = m["pv_name"]
```

Comment:
constants.

Reply:
Similar comments. Do we declare constants for every pv field?

Reply:
yes